### PR TITLE
You can now recharge labellers with paper, also, you can label glasses and surgery tools now.

### PR DESCRIPTION
**Pull request by F-OS**
This makes it easier to distribute a shitload of pills to marines on deployment with less of a struggle to tell them OK THIS DOES THIS.

### CODE REVIEW

**Comment by ninjanomnom** [comment]
Label the container...? Alternatively port how we handle pill naming in tg

**Comment by F-OS** [comment]
@ninjanomnom It just felt like a unneeded restriction to me. So i decided to remove it, do you see any issues with this?

**Comment by ninjanomnom** [comment]
I'd rather not remove the consumable labels aspect of it at all

**Comment by Rohesie** [comment]
Obscuring what pills do is kind of intended in this context. Anyone trained in medicine can know what the contents of the pills are, as odd as that sounds.

**Comment by F-OS** [comment]
I brought back the limit to labels and now you can toss paper into them.

**Comment by F-OS** [comment]
Remember kids, don't copy+paste code without running through it and making sure it works.

File: code/game/objects/items/tools/misc_tools.dm [hunk 1]
```diff
@@ -12,43 +12,34 @@
...
-	if(!labels_left)
-		to_chat(user, "<span class='notice'>No labels left.</span>")
-		return
 	if(!label || !length(label))
 		to_chat(user, "<span class='notice'>No text set.</span>")
 		return
 	if(length(A.name) + length(label) > 64)
 		to_chat(user, "<span class='notice'>Label too big.</span>")
 		return
-	if(isliving(A))
```
Comment (Rohesie) [inline code/game/objects/items/tools/misc_tools.dm]: Do we want to be able to label people? :thinking:

Comment (ninjanomnom) [reply code/game/objects/items/tools/misc_tools.dm]: No we don't, or really any mob.

File: code/game/objects/items/tools/misc_tools.dm [hunk 2]
```diff
@@ -12,43 +12,34 @@
 	icon_state = "labeler0"
 	item_state = "flight"
 	var/label = null
-	var/labels_left = 50
 	var/mode = 0	//off or on.
-
+	var/wait = 0
+	var/labels = 50
```
Comment (ninjanomnom) [inline code/game/objects/items/tools/misc_tools.dm]: Use the old var name

File: code/game/objects/items/tools/misc_tools.dm [hunk 3]
```diff
@@ -12,43 +12,34 @@
...
-	if(isliving(A))
-		to_chat(user, "<span class='notice'>You can't label living beings.</span>")
-		return
-	if(istype(A, /obj/item/reagent_container/glass))
-		to_chat(user, "<span class='notice'>The label will not stick to [A]. Use a pen instead.</span>")
-		return
-	if(istype(A, /obj/item/tool/surgery))
-		to_chat(user, "<span class='notice'>That wouldn't be sanitary.</span>")
-		return
+	if(world.time < wait + 10)
```
Comment (ninjanomnom) [inline code/game/objects/items/tools/misc_tools.dm]: A cooldown isn't needed

File: code/game/objects/items/tools/misc_tools.dm [hunk 4]
```diff
@@ -65,6 +56,12 @@
 		to_chat(user, "<span class='notice'>You turn off \the [src].</span>")
 
 
+/obj/item/tool/hand_labeler/attackby(obj/item/I, mob/user, params)
+	..()
+	if(istype(I, /obj/item/paper))
+		to_chat(user, "<span class='notice'>You insert [I] into [src].</span>")
+		cdel(I)
+		labels = min((labels+5),30)	//Yes, it's capped at its initial value
```
Comment (ninjanomnom) [inline code/game/objects/items/tools/misc_tools.dm]: `min(labels+5, initial(labels))`

File: code/game/objects/items/tools/misc_tools.dm [hunk 5]
```diff
@@ -12,43 +12,34 @@
...
-		return
-	if(istype(A, /obj/item/reagent_container/glass))
-		to_chat(user, "<span class='notice'>The label will not stick to [A]. Use a pen instead.</span>")
-		return
-	if(istype(A, /obj/item/tool/surgery))
-		to_chat(user, "<span class='notice'>That wouldn't be sanitary.</span>")
-		return
+	if(world.time < wait + 10)
+		to_chat(user, "<span class='notice'>The print heads are still cleaning, if you print the label right now it'll smear, you wouldn't want that? Would you?</span>")
+	if(!labels || labels == 0)
```
Comment (ninjanomnom) [inline code/game/objects/items/tools/misc_tools.dm]: 0 is a falsey value, the second part of this is unnecessary

File: code/game/objects/items/tools/misc_tools.dm [hunk 6]
```diff
@@ -12,43 +12,30 @@
 	icon_state = "labeler0"
 	item_state = "flight"
 	var/label = null
-	var/labels_left = 50
 	var/mode = 0	//off or on.
-
+	var/labels_left = 50
```
Comment (ninjanomnom) [inline code/game/objects/items/tools/misc_tools.dm]: You still have changes here and honestly removing the space before new definitions just makes things harder to read

File: code/game/objects/items/tools/misc_tools.dm [hunk 7]
```diff
@@ -12,43 +12,30 @@
...
-	if(istype(A, /obj/item/reagent_container/glass))
-		to_chat(user, "<span class='notice'>The label will not stick to [A]. Use a pen instead.</span>")
-		return
-	if(istype(A, /obj/item/tool/surgery))
-		to_chat(user, "<span class='notice'>That wouldn't be sanitary.</span>")
-		return
-	if(isturf(A))
+	if(labels_left == 0)
+		to_chat(user, "<span class='notice'>You've run out of labelling paper, feed some paper into it.</span>")
+	if(isturf(A) || isliving(A))
```
Comment (ninjanomnom) [inline code/game/objects/items/tools/misc_tools.dm]: Do what we do in tg, `ismob(A)`

File: code/game/objects/items/tools/misc_tools.dm [hunk 8]
```diff
@@ -12,43 +12,30 @@
...
-		to_chat(user, "<span class='notice'>You can't label living beings.</span>")
-		return
-	if(istype(A, /obj/item/reagent_container/glass))
-		to_chat(user, "<span class='notice'>The label will not stick to [A]. Use a pen instead.</span>")
-		return
-	if(istype(A, /obj/item/tool/surgery))
-		to_chat(user, "<span class='notice'>That wouldn't be sanitary.</span>")
-		return
-	if(isturf(A))
+	if(labels_left == 0)
```
Comment (ninjanomnom) [inline code/game/objects/items/tools/misc_tools.dm]: `if(!labels_left)` 0 is a falsey value

File: code/game/objects/items/tools/misc_tools.dm [hunk 9]
```diff
@@ -12,43 +12,30 @@
...
+	if(isturf(A) || isliving(A))
 		to_chat(user, "<span class='notice'>The label won't stick to that.</span>")
 		return
 
 	user.visible_message("<span class='notice'>[user] labels [A] as \"[label]\".</span>", \
 						 "<span class='notice'>You label [A] as \"[label]\".</span>")
 	A.name = "[A.name] ([label])"
-	labels_left--
-
+	labels_left--
```
Comment (ninjanomnom) [inline code/game/objects/items/tools/misc_tools.dm]: Again you're removing extra space that makes things easier to read

File: code/game/objects/items/tools/misc_tools.dm [hunk 10]
```diff
@@ -12,43 +12,30 @@
 	icon_state = "labeler0"
 	item_state = "flight"
 	var/label = null
-	var/labels_left = 50
 	var/mode = 0	//off or on.
-
+	var/labels_left = 50
 /obj/item/tool/hand_labeler/afterattack(atom/A, mob/user as mob, proximity)
 	if(!proximity) return
```
Comment (ninjanomnom) [inline code/game/objects/items/tools/misc_tools.dm]: Fix this while you're here

File: code/game/objects/items/tools/misc_tools.dm [hunk 11]
```diff
@@ -65,6 +52,12 @@
 		to_chat(user, "<span class='notice'>You turn off \the [src].</span>")
 
 
+/obj/item/tool/hand_labeler/attackby(obj/item/I, mob/user, params)
+	..()
```
Comment (ninjanomnom) [inline code/game/objects/items/tools/misc_tools.dm]: `. = ..()`

Comment (F-OS) [reply code/game/objects/items/tools/misc_tools.dm]: Is there any difference between the two?

Comment (ninjanomnom) [reply code/game/objects/items/tools/misc_tools.dm]: It assigns the value of the parent call to `.`

**Pull request merged**